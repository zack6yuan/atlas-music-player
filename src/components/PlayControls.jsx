import previous from "../assets/previous.svg"
import play from "../assets/play.svg"
import next from "../assets/next.svg";
import shuffle from "../assets/shuffle.svg";

export default function PlayControls() {
    return (
        <div className="play-controls flex 2xl:gap-18 xl:gap-17 lg:gap-17 md:gap-11">
            <p>1x</p>
            <img src={ previous } alt="previous-icon" class=""/>
            <img src={ play } alt="play-icon" class=""/>
            <img src={ next } alt="next-icon" class=""/>
            <img src={ shuffle } alt="shuffle-icon" class=""/>
        </div>
    )
}