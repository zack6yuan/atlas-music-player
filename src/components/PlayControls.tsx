import { useState } from 'react';

import play from "../assets/play.svg";
import pause from "../assets/pause.svg";
import next from "../assets/next.svg";
import shuffle from "../assets/shuffle.svg";
import shuffleClicked from "../assets/shuffle-clicked.svg";

export default function PlayControls( {prevImage, startSong}) {
  const [image, setImage] = useState(shuffle);
  const [playing, setPlaying] = useState(play);
  const [speed, setSpeed] = useState("1x");

  const changeImage = () => {
    if (image === shuffle) {
      setImage(shuffleClicked);
    } else {
      setImage(shuffle);
    }
  };

  const setPlay = () => {
    if (playing === pause) {
        setPlaying(play);
        startSong();
    } else {
        setPlaying(pause);
    }
  }

  const setSongSpeed = ()  => {
    if (speed === "1x") {
        setSpeed("2x");
    } else if (speed === "2x") {
        setSpeed("0.5x");
    } else if (speed === "0.5x") {
        setSpeed("1x");
    }
  }

  return (
    <div className="play-controls flex sm:gap-22 md:gap-8 lg:gap-14 lg:ml-2 lg:-mt-4 xl:gap-14.5 2xl:gap-14.5">
      <p onClick={ setSongSpeed } className="cursor-pointer select-none sm:mt-5 w-3 desc-text">{ speed }</p>
      <img src={ prevImage } alt="previous-icon" className="cursor-pointer" />
      <img src={ playing } alt="play-icon" className="rounded-lg border-3 p-4 cursor-pointer w-15 h-15" onClick={ setPlay }/>
      <img src={ next } alt="next-icon" className="cursor-pointer" />
      <img src={ image } alt="shuffle-icon" className="cursor-pointer" onClick={ changeImage }/>
    </div>
  );
}
