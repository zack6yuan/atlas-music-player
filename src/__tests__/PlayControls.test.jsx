import { render, screen, fireEvent } from "@testing-library/react";
import { expect, test } from "vitest";
import PlayControls from "../components/PlayControls"

test("Play Controls render correctly", () => {
    const { container } = render(
        <PlayControls />
    );
    expect(container).toMatchSnapshot();
});