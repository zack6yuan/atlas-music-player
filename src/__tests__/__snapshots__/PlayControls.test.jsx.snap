// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Play Controls render correctly 1`] = `
<div>
  <div
    class="play-controls flex sm:gap-22 md:gap-8 lg:gap-14 lg:ml-2 lg:-mt-4 xl:gap-14.5 2xl:gap-14.5"
  >
    <p
      class="cursor-pointer select-none sm:mt-5 w-3 desc-text"
      data-testid="speed-setter"
    >
      1x
    </p>
    <img
      alt="previous-icon"
      class="cursor-pointer"
    />
    <img
      alt="play-icon"
      class="rounded-lg border-3 p-4 cursor-pointer w-15 h-15"
      src="data:image/svg+xml,%3csvg%20width='16'%20height='20'%20viewBox='0%200%2016%2020'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M1%201L15%2010L1%2019V1Z'%20fill='black'%20stroke='black'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3c/svg%3e"
    />
    <img
      alt="next-icon"
      class="cursor-pointer"
      src="data:image/svg+xml,%3csvg%20width='24'%20height='24'%20viewBox='0%200%2024%2024'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%20clip-path='url(%23clip0_263_108)'%3e%3cpath%20d='M12%2019L21%2012L12%205V19Z'%20fill='black'%20stroke='black'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3cpath%20d='M1%2019L10%2012L1%205V19Z'%20fill='black'%20stroke='black'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3c/g%3e%3cdefs%3e%3cclipPath%20id='clip0_263_108'%3e%3crect%20width='24'%20height='24'%20fill='white'/%3e%3c/clipPath%3e%3c/defs%3e%3c/svg%3e"
    />
    <img
      alt="shuffle-icon"
      class="cursor-pointer"
      src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='24'%20height='24'%20viewBox='0%200%2024%2024'%20fill='none'%20stroke='currentColor'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'%20class='lucide%20lucide-shuffle-icon%20lucide-shuffle'%3e%3cpath%20d='m18%2014%204%204-4%204'/%3e%3cpath%20d='m18%202%204%204-4%204'/%3e%3cpath%20d='M2%2018h1.973a4%204%200%200%200%203.3-1.7l5.454-8.6a4%204%200%200%201%203.3-1.7H22'/%3e%3cpath%20d='M2%206h1.972a4%204%200%200%201%203.6%202.2'/%3e%3cpath%20d='M22%2018h-6.041a4%204%200%200%201-3.3-1.8l-.359-.45'/%3e%3c/svg%3e"
    />
  </div>
</div>
`;

exports[`Speed button changes speed 1`] = `
<div>
  <div
    class="play-controls flex sm:gap-22 md:gap-8 lg:gap-14 lg:ml-2 lg:-mt-4 xl:gap-14.5 2xl:gap-14.5"
  >
    <p
      class="cursor-pointer select-none sm:mt-5 w-3 desc-text"
      data-testid="speed-setter"
    >
      1x
    </p>
    <img
      alt="previous-icon"
      class="cursor-pointer"
    />
    <img
      alt="play-icon"
      class="rounded-lg border-3 p-4 cursor-pointer w-15 h-15"
      src="data:image/svg+xml,%3csvg%20width='16'%20height='20'%20viewBox='0%200%2016%2020'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M1%201L15%2010L1%2019V1Z'%20fill='black'%20stroke='black'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3c/svg%3e"
    />
    <img
      alt="next-icon"
      class="cursor-pointer"
      src="data:image/svg+xml,%3csvg%20width='24'%20height='24'%20viewBox='0%200%2024%2024'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%20clip-path='url(%23clip0_263_108)'%3e%3cpath%20d='M12%2019L21%2012L12%205V19Z'%20fill='black'%20stroke='black'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3cpath%20d='M1%2019L10%2012L1%205V19Z'%20fill='black'%20stroke='black'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3c/g%3e%3cdefs%3e%3cclipPath%20id='clip0_263_108'%3e%3crect%20width='24'%20height='24'%20fill='white'/%3e%3c/clipPath%3e%3c/defs%3e%3c/svg%3e"
    />
    <img
      alt="shuffle-icon"
      class="cursor-pointer"
      src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='24'%20height='24'%20viewBox='0%200%2024%2024'%20fill='none'%20stroke='currentColor'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'%20class='lucide%20lucide-shuffle-icon%20lucide-shuffle'%3e%3cpath%20d='m18%2014%204%204-4%204'/%3e%3cpath%20d='m18%202%204%204-4%204'/%3e%3cpath%20d='M2%2018h1.973a4%204%200%200%200%203.3-1.7l5.454-8.6a4%204%200%200%201%203.3-1.7H22'/%3e%3cpath%20d='M2%206h1.972a4%204%200%200%201%203.6%202.2'/%3e%3cpath%20d='M22%2018h-6.041a4%204%200%200%201-3.3-1.8l-.359-.45'/%3e%3c/svg%3e"
    />
  </div>
</div>
`;
